pandas-stubs/core/indexes/interval.pyi

```diff
@@ -216,46 +250,155 @@ class IntervalIndex(IntervalMixin, ExtensionIndex, Generic[_IntervalT]):
+@overload
+def interval_range(  # type: ignore[misc]
+    start: int = ...,
+    end: int = ...,
+    periods: int | None = ...,
+    freq: int | None = ...,
+    name: Hashable = ...,
+    closed: IntervalClosedType = ...,
+) -> IntervalIndex[Interval[int]]: ...
 
+# Overlaps since int is a subclass of float
+@overload
+def interval_range(  # pyright: reportOverlappingOverload=false
+    start: int,
+    *,
+    end: None = ...,
+    periods: int | None = ...,
+    freq: int | None = ...,
+    name: Hashable = ...,
+    closed: IntervalClosedType = ...,
+) -> IntervalIndex[Interval[int]]: ...
+@overload
+def interval_range(  # pyright: reportOverlappingOverload=false
+    *,
+    start: None = ...,
+    end: int,
+    periods: int | None = ...,
+    freq: int | None = ...,
+    name: Hashable = ...,
+    closed: IntervalClosedType = ...,
+) -> IntervalIndex[Interval[int]]: ...
+@overload
+def interval_range(
+    start: float = ...,
+    end: float = ...,
+    periods: int | None = ...,
+    freq: int | None = ...,
+    name: Hashable = ...,
+    closed: IntervalClosedType = ...,
+) -> IntervalIndex[Interval[float]]: ...
+@overload
+def interval_range(
+    start: float,
+    *,
+    end: None = ...,
+    periods: int | None = ...,
+    freq: int | None = ...,
+    name: Hashable = ...,
+    closed: IntervalClosedType = ...,
+) -> IntervalIndex[Interval[float]]: ...
 @overload
 def interval_range(
-    start: int | float | None = ...,
-    end: int | float | None = ...,
+    *,
+    start: None = ...,
+    end: float,
     periods: int | None = ...,
     freq: int | None = ...,
     name: Hashable = ...,
     closed: IntervalClosedType = ...,
-) -> IntervalIndex: ...
+) -> IntervalIndex[Interval[float]]: ...
+@overload
+def interval_range(
+    start: _TimestampLike,
+    end: _TimestampLike = ...,
+    periods: int | None = ...,
+    freq: str | BaseOffset | None = ...,
+    name: Hashable = ...,
+    closed: IntervalClosedType = ...,
+) -> IntervalIndex[Interval[pd.Timestamp]]: ...
+@overload
+def interval_range(
+    *,
+    start: None = ...,
+    end: _TimestampLike,
+    periods: int | None = ...,
+    freq: str | BaseOffset | None = ...,
+    name: Hashable = ...,
+    closed: IntervalClosedType = ...,
+) -> IntervalIndex[Interval[pd.Timestamp]]: ...
+@overload
+def interval_range(
+    start: _TimestampLike,
+    *,
+    end: None = ...,
+    periods: int | None = ...,
+    freq: str | BaseOffset | None = ...,
+    name: Hashable = ...,
+    closed: IntervalClosedType = ...,
+) -> IntervalIndex[Interval[pd.Timestamp]]: ...
+@overload
+def interval_range(
+    *,
+    start: None = ...,
+    end: _TimedeltaLike,
+    periods: int | None = ...,
+    freq: str | BaseOffset | None = ...,
+    name: Hashable = ...,
+    closed: IntervalClosedType = ...,
+) -> IntervalIndex[Interval[pd.Timestamp]]: ...
 @overload
 def interval_range(
-    start: DatetimeLike | None = ...,
-    end: DatetimeLike | None = ...,
+    start: _TimedeltaLike,
```

Comment:
shouldn't there be a more complete set of overloads for `_TimedeltaLIke` just like there are for the others (`int`, `float`, `_TimestampLike`)

Reply:
Added the one missing, and some tests.